docs/tutorials/mongoid-documents.txt

```diff
@@ -215,23 +212,23 @@ An example usage is shown below:
   post.status
   # => :hello
 
-Values which are neither strings nor symbols will be converted to strings
-(via ``to_s``) during serialization. Values which are neither strings nor
-symbols in MongoDB will be converted to strings and then to symbols during
-deserialization.
+All non-string values will be stringified upon being sent to the database (via ``to_s``), and
+all values will be converted to symbols when returned to the application. Values that cannot be
+converted directly to symbols, such as integers and arrays, will first be converted to strings and
+then symbols before being returned to the application.
 
 For example, setting an integer as ``status``:
 
 .. code-block:: ruby
 
   post = Post.new(status: 42)
-  post.status # => :"42"
-
-Values which are stored in the database as symbols will be returned to the
-application as symbols, but will be stored as strings at the next save.
-This permits transparent lazy migration from fields that currently store either
-strings or BSON symbols in MongoDB to the ``StringifiedSymbol`` field type.
+  post.status
+  # => :"42"
 
+If the ``StringifiedSymbol`` type is applied to a field that contains ``BSON::Symbol``
+data, the documents will be stored with ``String`` values instead of the deprecated ``BSON::Symbol``
```

Comment:
It is not technically correct to speak about data stored in the database in terms of Ruby types. The correct language to use would refer to BSON types. Thus, "BSON symbol" and not `BSON::Symbol` for example.